src/lib/contracts/utils.ts

```diff
@@ -46,3 +46,59 @@ export async function getGasLimit(
 
   return (gasLimit * 11n) / 10n; // add a 10% buffer
 }
+
+export function getBestNonce(providers: Wallet[]): Promise<number> {
+  const MAX_NONCE_NEEDED = 3;
+  const MAX_WAIT = 5000;
+  const ONE_MORE_WAIT = 1000;
+
+  return new Promise(async (resolve, reject) => {
+    const results: number[] = [];
+    let resolved = false;
+
+    const handleResolve = () => {
+      resolved = true;
```

Comment:
a small suggestion, you can clear timeout here, so there will be no need to track `resolved` and check it inside `setResolveTimeout`

Reply:
discussed it dm, no need to do this